octobot/src/server/github_handler.rs

```diff
@@ -409,22 +455,24 @@ impl GithubEventHandler {
         participants.push(pull_request.user().clone());
         // add team members
         let teams = pull_request.teams();
-        let futures = teams
-            .iter()
-            .map(|t| self.github_session.get_team_members(&t.url));
-        let team_members = future::join_all(futures).await;
-        participants.extend(
-            team_members
-                .into_iter()
-                .filter_map(|r| match r {
-                    Ok(r) => Some(r),
+        for t in teams {
+            let team_members = self.team_members_cache.get(&t.slug);
+            if team_members != None {
+                participants.append(&mut team_members.unwrap().clone());
+            } else {
+                let team_members = self.github_session.get_team_members(&t.url).await;
+                match team_members {
+                    Ok(m) => {
+                        participants.append(&mut m.clone());
```

Comment:
i think here this can do something like this to avoid cloning:

```suggestion
                        participants.extend(m.into_iter());
```

Reply:
I still had to clone since `m` was used again after this call.